amundsen_application/static/js/ducks/tags/api/v0.ts

```diff
@@ -0,0 +1,52 @@
+import axios, { AxiosResponse } from 'axios';
+
+import { sortTagsAlphabetical } from 'ducks/utilMethods';
+import { ResourceType, Tag } from 'interfaces';
+import { getTableTagsFromResponseData } from 'ducks/tableMetadata/api/helpers';
+import { API_PATH, TableDataAPI } from 'ducks/tableMetadata/api/v0';
+import { GetDashboardAPI } from 'ducks/dashboard/api/v0';
+import { getDashboardTagsFromResponseData } from 'ducks/dashboard/api/helpers';
+
+export type AllTagsAPI = {
+  msg: string;
+  tags: Tag[];
+};
+
+export function getAllTags() {
+  return axios.get('/api/metadata/v0/tags').then((response: AxiosResponse<AllTagsAPI>) => {
+    return response.data.tags.sort(sortTagsAlphabetical);
+  })
+};
+
+export function getTags(resourceType, uriKey: string) {
+  if (resourceType === ResourceType.table) {
+    return axios.get(`${API_PATH}/table?key=${uriKey}`)
+    .then((response: AxiosResponse<TableDataAPI>) => {
+      return getTableTagsFromResponseData(response.data);
+    });
+  }
+  if (resourceType === ResourceType.dashboard) {
+    return axios.get(`${API_PATH}/dashboard?uri=${uriKey}`)
+    .then((response: AxiosResponse<GetDashboardAPI>) => {
+      return getDashboardTagsFromResponseData(response.data)
+    });
+  }
+}
+
+/* TODO: Typing this method generates redux-saga related type errors that needs more dedicated debugging */
+export function updateTableTag(tagObject, resourceType: ResourceType, uriKey: string) {
+  let url = "";
+  if (resourceType === ResourceType.table) {
```

Comment:
I agree, but I think I'll leave this for a follow up. I wonder if it also makes sense to approach the search APIs in a similar way. Probably need to spend more time thinking about it first, however.

Reply:
For the search APIs one thing that makes me hold off in that area is that the search service might also eventually refactor endpoints so that the same endpoints can be used for all resources to make things scale better as we add more resource or the community adds more resources to their custom versions.  I'd want to see exactly what that will look like before messing anything up here.